src/document/Document.js

```diff
@@ -698,7 +698,24 @@ define(function (require, exports, module) {
     Document.prototype.isUntitled = function () {
         return this.file instanceof InMemoryFile;
     };
+
+    /**
+     * Set lineEnding of the document. Change Text in the process.
+     */
+    Document.prototype.setLineEndings = function (lineEndings) {
+        this._lineEndings = lineEndings;
+        this.setText(FileUtils.translateLineEndings(this.getText(true), lineEndings));
```

Comment:
That's just a side-effect: a no-op change still marks the document dirty, and you can only save documents when they're dirty.  If you remove this line, it will still work fine if you make any edits to the document (marking it dirty) before saving.

It does sound like this needs to be integrated with the dirty flag to have intuitive behavior, since it'd definitely be confusing if even explicitly saving wouldn't persist the change.  I don't think this is the right way to achieve that though.